tools/accuracy_checker/openvino/tools/accuracy_checker/metrics/background_matting.py

```diff
@@ -0,0 +1,173 @@
+        else:
+            raise ValueError('Unsupported format of image!')
+
+    def get_annotation(self, annotation):
+        return self.process_func(annotation.value)
+
+    def get_prediction(self, prediction):
+        return prediction.value[self.prediction_source]
+
+    def evaluate(self, annotations, predictions):
+        return sum(self.results) / len(self.results)
+
+    def reset(self):
+        self.results = []
+
+    def configure(self):
+        process_type = self.get_value_from_config('process_type')
+        self.process_func = self.prepare_pha if process_type == 'alpha' else self.prepare_fgr
+        self.reset()
+
+    @classmethod
+    def get_common_meta(cls):
+        return {'target': 'higher-worse'}
+
+
+class MeanOfAbsoluteDifference(BaseBackgroundMattingMetrics):
+    __provider__ = 'mean_of_absolute_difference'
+
+    def update(self, annotation, prediction):
+        pred = self.get_prediction(prediction)
+        gt = self.get_annotation(annotation)
+        value = np.mean(abs(pred - gt)) * 1e3
+        self.results.append(value)
+        return value
+
+
+class SpatialGradient(BaseBackgroundMattingMetrics):
+    __provider__ = 'spatial_gradient'
+
+    def update(self, annotation, prediction):
+        pred = self.get_prediction(prediction)
+        gt = self.get_annotation(annotation)
+        gt_grad = self.gauss_gradient(gt)
+        pred_grad = self.gauss_gradient(pred)
+        value = np.sum((gt_grad - pred_grad) ** 2) / 1000
+        self.results.append(value)
+        return value
+
+    def gauss_gradient(self, img):
+        img_filtered_x = convolve(img, self.filter_x[:, ::-1], mode='reflect')
+        img_filtered_y = convolve(img, self.filter_y[::-1, :], mode='reflect')
+        return np.sqrt(img_filtered_x ** 2 + img_filtered_y ** 2)
+
+    def gauss_filter(self, sigma, epsilon=1e-2):
+        half_size = np.ceil(sigma * np.sqrt(-2 * np.log(np.sqrt(2 * np.pi) * sigma * epsilon)))
+        size = np.int(2 * half_size + 1)
+        # create filter in x axis
+        filter_x = np.zeros((size, size))
+        for i in range(size):
+            for j in range(size):
+                filter_x[i, j] = self.gaussian(i - half_size, sigma) * self.dgaussian(
+                    j - half_size, sigma)
+        # normalize filter
+        norm = np.sqrt((filter_x ** 2).sum())
+        filter_x = filter_x / norm
+        filter_y = np.transpose(filter_x)
+        return filter_x, filter_y
+
+    @staticmethod
+    def gaussian(x, sigma):
+        return np.exp(-x ** 2 / (2 * sigma ** 2)) / (sigma * np.sqrt(2 * np.pi))
+
+    def dgaussian(self, x, sigma):
+        return -x * self.gaussian(x, sigma) / sigma ** 2
+
+    def configure(self, sigma=1.4):
+        super().configure()
+        self.filter_x, self.filter_y = self.gauss_filter(sigma)
+        self.reset()
+
+
+class MeanSquaredErrorWithMask(BaseBackgroundMattingMetrics):
+    __provider__ = 'mean_squared_error'
```

Comment:
```suggestion
    __provider__ = 'mse'
```

Reply:
`mse` is already used. Renamed to `mse_with_mask` like used class name.